src/MiddlewareRegistry.ts

```diff
@@ -33,7 +33,7 @@ export class MiddlewareRegistry<R extends MiddlewareRequest> {
         middleware: MiddlewareFunction | MiddlewareFunction[],
         config?: Omit<MiddlewareConfig, 'middleware'>
     ) {
-        this.registry.set(route, { ...config, middleware: middleware });
+        this.registry.set(this.serializeToRegistryKey(route, config?.methods), { ...config, middleware: middleware });
```

Comment:
Nit: Unrelated to this MR, but I think we can reduce `middleware: middleware` to just `middleware`
```suggestion
        this.registry.set(this.serializeToRegistryKey(route, config?.methods), { ...config, middleware });
```

Reply:
Good call. Done.